src/Rector/ClassMethod/AddArgumentDefaultValueRector.php

```diff
@@ -0,0 +1,117 @@
+use Rector\Core\Rector\AbstractRector;
+use Rector\Laravel\ValueObject\AddArgumentDefaultValue;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see \Rector\Laravel\Tests\Rector\ClassMethod\AddArgumentDefaultValueRector\AddArgumentDefaultValueRectorTest
+ */
+final class AddArgumentDefaultValueRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    /**
+     * @var string
+     */
+    public const ADDED_ARGUMENTS = 'added_arguments';
+
+    /**
+     * @var AddArgumentDefaultValue[]
+     */
+    private array $addedArguments = [];
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Adds default value for arguments in defined methods.',
+            [
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function someMethod($value)
+    {
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function someMethod($value = false)
+    {
+    }
+}
+CODE_SAMPLE
+                    ,
+                    [
+                        self::ADDED_ARGUMENTS => [
+                            new AddArgumentDefaultValue('SomeClass', 'someMethod', 0, false),
+                        ],
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ClassMethod
+    {
+        foreach ($this->addedArguments as $addedArgument) {
+            if (! $this->nodeTypeResolver->isObjectType($node, $addedArgument->getObjectType())) {
+                continue;
+            }
+
+            if (! $this->isName($node->name, $addedArgument->getMethod())) {
+                continue;
+            }
+
+            if (! isset($node->params[$addedArgument->getPosition()])) {
+                continue;
+            }
+
+            $position = $addedArgument->getPosition();
+            $param = $node->params[$position];
+
+            if ($param->default !== null) {
+                continue;
+            }
+
+            $node->params[$position] = new Param($param->var, BuilderHelpers::normalizeValue(
+                $addedArgument->getDefaultValue()
+            ));
+        }
+
+        return $node;
+    }
+
+    /**
+     * @param array<string, AddArgumentDefaultValue[]> $configuration
+     */
+    public function configure(array $configuration): void
+    {
+        $addedArguments = $configuration[self::ADDED_ARGUMENTS] ?? [];
+        Assert::allIsInstanceOf($addedArguments, AddArgumentDefaultValue::class);
+        $this->addedArguments = $addedArguments;
+    }
+}
```

Comment:
Just noticted this... maybe you can use: https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#argumentadderrector

Reply:
I tried `ArgumentAdderRector` before, but it only works when there is no argument in the position. The problem here is that there is already an argument without default value in this position, and it needs to be optional. 

Here is the configuration that didn't work before:
```php
$services->set(ArgumentAdderRector::class)
    ->call('configure', [[
        ArgumentAdderRector::ADDED_ARGUMENTS => ValueObjectInliner::inline([
            new ArgumentAdder('Illuminate\Contracts\Events\Dispatcher', 'listen', 1, 'listener', null),
        ]),
    ]]);
```

The option `added_arguments` here seems to be ambiguous.
https://github.com/rectorphp/rector-laravel/blob/b58ac1f66ac9723b5a2eb4b5d7a218d0404d1b47/src/Rector/ClassMethod/AddArgumentDefaultValueRector.php#L23-L26

Reply:
> I tried ArgumentAdderRector before, but it only works when there is no argument in the position. 

I see. This feature should be supported with that rule too. E.g. with `*ANY*` constant in this rule:
https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#replaceargumentdefaultvaluerector

The split packages like rector-laravel should only contain Laravel-specific rules. This can be re-used in generic way, so it should located in rector-src. Could you move it there?

Reply:
> I see. This feature should be supported with that rule too.

Created a draft PR in case I misunderstood what you mean rectorphp/rector-src#837.

> E.g. with `*ANY*` constant in this rule:
> https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#replaceargumentdefaultvaluerector

I'm not sure about the scope of this constant. `ReplaceArgumentDefaultValueRector` only handles `MethodCall` and `StaticCall` now, replace any value before with value after?